core/src/main/java/com/linecorp/armeria/common/thrift/ThriftCompletableFuture.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.thrift;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.thrift.async.AsyncMethodCallback;
+
+/**
+ * A {@link CompletableFuture} that can be passed in as an {@link AsyncMethodCallback}
+ * when making an asynchronous client-side Thrift RPC.
+ */
+public class ThriftCompletableFuture<T> extends CompletableFuture<T> implements AsyncMethodCallback<T> {
+    /**
+     * Returns a new instance that has its value set immediately.
+     */
+    public static <T> ThriftCompletableFuture<T> successful(T value) {
+        ThriftCompletableFuture<T> future = new ThriftCompletableFuture<T>();
+        future.onComplete(value);
+        return future;
+    }
+
+    /**
+     * Returns a new instance that has an exception set immediately.
+     */
+    public static <T> ThriftCompletableFuture<T> failed(Exception e) {
+        ThriftCompletableFuture<T> future = new ThriftCompletableFuture<>();
+        future.onError(e);
+        return future;
+    }
+
```

Comment:
oh, I completely forgot about parent class API. How about add utility class (for example, ThriftFutures) and move these static factory methods to this class? Like this:
```java
class ThriftFutures {
  public static <T> ThriftListenableFuture<T> successfulListenableFuture(T value) {...}
 // or public static <T> ThriftListenableFuture<T> successfulThriftListenableFuture(T value) {...} 
  public static <T> ThriftListenableFuture<T> failedListenableFuture(Exception e) {...}

  public static <T> ThriftCompletableFuture<T> successfulCompletedFuture(T value) {...}
  public static <T> ThriftCompletableFuture<T> failedCompletedFuture(Exception e) {...}
}
```